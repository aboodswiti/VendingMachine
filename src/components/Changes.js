import React from "react";

const Changes = ({ allMoney, availableChange, returnedChange }) => {
  // Display Total money & availableChange inside VM
  return (
    <div className="changes">
      <div>
        {Object.keys(returnedChange).length ? <p>Change : </p> : null}
        {Object.entries(returnedChange).map(([key, val]) => (
          <p key={key}>
            ${key}: {val}
          </p>
        ))}
      </div>

      <p>
        Available
        <br />
        Change :
      </p>
      <h2>${allMoney}</h2>

      <h4> $Type : #Amount </h4>
      {Object.entries(availableChange).map(([key, val]) => (
        <h3 key={key}>
          ${key}: {val}
        </h3>
      ))}
    </div>
  );
};

export default Changes;
