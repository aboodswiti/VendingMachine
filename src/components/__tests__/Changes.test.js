import React from "react";
import {cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";


import Changes from "../Changes";
afterEach(() => {
  cleanup();
});



  test('Changes,Should Matches snapshot', ()=> {
    const returnedChange = { 50:1}
    const availableChange = {1: 17,
      20: 4,
      50: 3,
      0.1: 8,
      0.2: 5,
      0.5: 4,}

const tree = renderer.create(<Changes returnedChange={returnedChange} availableChange={availableChange} />).toJSON();

expect(tree).toMatchSnapshot();
} );