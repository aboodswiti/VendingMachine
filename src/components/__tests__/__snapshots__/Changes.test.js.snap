// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changes,Should Matches snapshot 1`] = `
<div
  className="changes"
>
  <div>
    <p>
       Your Changes : 
    </p>
    <p>
      $
      50
      : 
      1
    </p>
  </div>
  <p>
    Available
    <br />
    Change :
  </p>
  <h2>
    $
  </h2>
  <h4>
     $Type : #Amount 
  </h4>
  <h3>
    1
    : 
    17
  </h3>
  <h3>
    20
    : 
    4
  </h3>
  <h3>
    50
    : 
    3
  </h3>
  <h3>
    0.1
    : 
    8
  </h3>
  <h3>
    0.2
    : 
    5
  </h3>
  <h3>
    0.5
    : 
    4
  </h3>
</div>
`;
